cmd/bucket-replication.go

```diff
@@ -293,29 +294,47 @@ func (r *replicationState) queueReplicaTask(oi ObjectInfo) {
 	}
 }
 
-var globalReplicationState *replicationState
+var (
+	globalReplicationState *replicationState
+	// TODO: currently keeping it conservative
+	// but eventually can be tuned in future,
+	// take only half the CPUs for replication
+	// conservatively.
+	globalReplicationConcurrent = runtime.NumCPU() / 2
```

Comment:
```suggestion
	globalReplicationConcurrent = runtime.GOMAXPROCS(0) / 2
```

That will make it match any custom setting. Not sure I would call half the CPUs as conservative ;)